lib/yggdrasil/player.ex

```diff
@@ -12,9 +13,8 @@ defmodule Yggdrasil.Player do
     Supervisor.add_player user_id, [owner_pid, push_msg]
   end
 
-  def run_cmd(user_id, cmd = %Message{type: :command}) do
-    player_pid = Registry.get_player user_id
-    GenServer.cast(player_pid, {:run_cmd, cmd})
+  def run_cmd(user_id, cmd) do
+    Command.execute cmd, user_id
```

Comment:
So I think the room should "execute" the command because I don't want to worry about a command succeeded but then the room saying "oh you failed because that doesn't exist anymore". Having the room execute the command then immediately handle the actions triggered means we don't have to worry about that. The command can know.

- Sean Copenhaver

> On Jan 28, 2016, at 3:24 PM, Matthew Philyaw <notifications@github.com> wrote:
> 
> In lib/yggdrasil/player.ex:
> 
> > @@ -12,9 +13,8 @@ defmodule Yggdrasil.Player do
> >      Supervisor.add_player user_id, [owner_pid, push_msg]
> >    end
> >  
> > -  def run_cmd(user_id, cmd = %Message{type: :command}) do
> > -    player_pid = Registry.get_player user_id
> > -    GenServer.cast(player_pid, {:run_cmd, cmd})
> > +  def run_cmd(user_id, cmd) do
> > +    Command.execute cmd, user_id
> That said, I think both agree that the room should handle the actual "execution" of the events and do agree that it would naturally serialize the actions.
> 
> All I was suggesting is that commands build the context and we dispatch the context object to the room and as it works through the actions it creates all the side effects etc.
> 
> —
> Reply to this email directly or view it on GitHub.
>